surface/include/pcl/surface/ball_pivoting.h

```diff
@@ -0,0 +1,323 @@
+     *        flipping between front surface and back surface is enabled (setAllowFlip)
+     * @param index index of triangle vertices, which are all one ball surface
+     * @param is_back_ball whether the pivoted ball is on back surface
+     * @return
+     */
+    boost::shared_ptr<Eigen::Vector3f>
+    getBallCenter (const bool is_back_first, std::vector<uint32_t> &index, bool &is_back_ball) const;
+  
+  public:
+    BallPivoting ():
+      radius_ (-1.0), 
+      is_allow_back_ball_ (false), 
+      is_allow_flip_ (false),
+      threshold_collinear_cos_ (cos (10.0 * M_PI / 180.0)), 
+      threshold_distance_near_ (1e-6)
+    {
+    }
+
+  
+    ~BallPivoting ()
+    {
+    }
+  
+    /**
+     * set the radius of ball
+     * @param radius
+     */
+    void
+    setSearchRadius (const double radius)
+    { radius_ = radius; }
+  
+    /**
+     * get the radius of ball
+     * @return
+     */
+    double
+    getSearchRadius () const
+    { return radius_; }
+  
+    /**
+     * set whether ball on the back surface is allowed. if this value is set to true, the ball only pivots on the
+     * direction of the normal vectors
+     * @param is_allow_back_ball
+     */
+    void
+    setAllowBackBall (const bool is_allow_back_ball)
+    { is_allow_back_ball_ = is_allow_back_ball; }
+  
+    /**
+     * get whether ball on the back surface is allowed
+     * @return
+     */
+    bool
+    getAllowBackBall () const
+    { return is_allow_back_ball_; }
+  
+    /**
+     * set whether flipping between front surface and back surface is allow. if not, the ball would only pivot
+     * so it stays on one side of the surface, depending on where the center of seed ball was found
+     * @param is_allow_flip
+     */
+    void
+    setAllowFlip (const bool is_allow_flip)
+    { is_allow_flip_ = is_allow_flip; }
+  
+    /**
+     * get whether flipping between front surface and back surface is allow
+     * @return
+     */
+    bool
+    getAllowFlip () const
+    { return is_allow_flip_; }
+  
+    /**
+     * Create the surface.
+     * @param output the resultant polygonal mesh
+     */
+    void
+    performReconstruction (pcl::PolygonMesh &output);
+  
+    /**
+     * Create the surface.
+     * @param points the vertex positions of the resulting mesh
+     * @param polygons the connectivity of the resulting mesh
+     */
+    void
+    performReconstruction (pcl::PointCloud<PointNT> &points,
+                           std::vector<pcl::Vertices> &polygons);
+
+    /**
+     * estimate one radius and set it as ball radius. num_sample_point points would be selected randomly in cloud,
+     * as least ratio_success of the sample points must have at least num_point_in_radius neighbors within
+     * the estimated radius.
+     * @param num_sample_point
+     * @param num_point_in_radius
+     * @param ratio_success
+     */
+    void
+    setSearchRadiusAutomatically (const int num_sample_point = 0, 
```

Comment:
In the `guess_search` function the default used to be 500. Is this a typo maybe?